### PR TITLE
Inbox filter

### PR DESCRIPTION
**Pull request by kntsoriano**
Closes https://github.com/exercism/v3-project-management/issues/97.
Closes https://github.com/exercism/v3-project-management/issues/88.

## Description
This implements the inbox track filter as a menu.

### CODE REVIEW

File: app/javascript/components/dropdowns/Dropdown.tsx [hunk 1]
```diff
@@ -20,123 +19,30 @@ export const Dropdown = ({
...
-          role="menu"
-          hidden={!open}
-        >
+      <div {...panelAttributes}>
+        <ul className={`${menuButton.className}-dropdown`} {...listAttributes}>
           {menuItems.map((item, i) => {
             return (
               <li
-                ref={(instance) => handleMenuItemMount(instance, i)}
                 key={item.html}
```
Comment (ErikSchierboom) [inline app/javascript/components/dropdowns/Dropdown.tsx]: I don't know much React, but would setting the key to the index result in less data in the HTML?

File: app/javascript/components/dropdowns/Dropdown.tsx [hunk 2]
```diff
@@ -20,123 +19,30 @@ export const Dropdown = ({
...
-      return
-    }
-
-    menuItemElementsRef.current[focusIndex].focus()
-  }, [open, focusIndex, buttonRef])
+  const {
+    buttonAttributes,
+    panelAttributes,
+    listAttributes,
+    itemAttributes,
```
Comment (ErikSchierboom) [inline app/javascript/components/dropdowns/Dropdown.tsx:26]: So clean!

File: app/javascript/components/mentoring/inbox/TrackList.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  value: string | null
+  setTrack: (value: string | null) => void
+}): JSX.Element => {
+  const {
+    buttonAttributes,
+    panelAttributes,
+    listAttributes,
+    itemAttributes,
+  } = useDropdown(tracks.length)
+  const selected = tracks.find((track) => track.slug === value) || tracks[0]
```
Comment (ErikSchierboom) [inline app/javascript/components/mentoring/inbox/TrackList.tsx:55]: Are these the tracks that a mentor is mentoring? I guess there will always be at least one?

**Review by ErikSchierboom — APPROVED** [review]

File: app/javascript/components/mentoring/inbox/TrackList.tsx (hunk 3 above)
Comment (iHiD) [reply app/javascript/components/mentoring/inbox/TrackList.tsx:55]: If there isn't then the attribute shouldn't appear at all. I'll discuss this with Tai.

**Comment by iHiD** [comment]
@kntsoriano Nice work :) I've added commits for styling. 

There's one bug I found, which is you don't seem to be able to select one of the options using the keyboard. Pressing space or enter just closes the dialog.

**Comment by kntsoriano** [comment]
@iHiD Fixed.

**Pull request merged**